float64 mappedX
float64 mappedY